providers/base/tests/test_gl_support.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# Written by:
+#   Hanhsuan Lee <hanhsuan.lee@canonical.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from gl_support import *
+from unittest.mock import patch, MagicMock
+import unittest
+
+
+class RemoveColorCode(unittest.TestCase):
+    """
+    This function should remove color code
+    """
+
+    unchanged = r"""
+OpenGL vendor string:   Intel
+OpenGL renderer string: Mesa Intel(R) Arc(tm) Graphics (MTL)
+OpenGL version string:4.6 (Compatibility Profile)Mesa 23.2.1-1ubuntu3.1~22.04.2
```

Comment:
It seems to me that the data format is a bit different with the output by unity_support_test utility, and I believe that's because you did not changes to fit black format check.
So I would suggest we place the original data in test_data directory, and use it in the unit test.

e.g. https://github.com/canonical/checkbox/blob/main/providers/base/tests/test_data/mdstat_intel_rst.txt

Reply:
This only thing changed in this line is removing spaces and I don't think this will impact the unit test correctness for removing color code.